tests/unit/test_artifactory_path.py

```diff
@@ -20,17 +22,40 @@
 
 class UtilTest(unittest.TestCase):
     def test_matrix_encode(self):
-        params = {"foo": "bar", "qux": "asdf"}
+        with mock.patch("urllib.parse.quote", new=mock.Mock(wraps=quote_original)) as q:
```

Comment:
Can we use `@mock.patch` decorator? To reduce indents 

Why do we need to patch `quote` with the same original `quote` function? I didn't get it, sorry

Reply:
Now I get it. We assert mock calls after.

For tests it doesn't matter wether we call original `quote` or somehow encoded params manually, we should assert against pure values - 
```python
assert deploy() == {"bar": ["bar bar"]}
assert deploy(quote=True) == {"bar": ["bar%20bar"]}
```

`quote` here is an implementation, and tests should not rely on this

Reply:
interesting because I was thinking of it the other way: that it doesn't matter what the pure values are because we don't care about how `quote` works, we trust its implementation, we just want to ensure we called it at the right times with the right parameters 🙃 (I left the value check in just to sort of be consistent with the other test cases)

If we implemented our own quote function, I would probably even do the same thing: I'd test that function's implementation separately, and in the tests for this `matrix_encode` function, I would be checking that it called the quote function.

But I think it just my opinion and neither is right or wrong so I am ok to change it if you prefer that.

Should I remove the mocking and call checks completely then?

Reply:
Yes, let's remove all mocks. 

>we trust its implementation

I've seen this approach few times and it can end up with something like this:
```python
def myfunction():
    obj = function1()
    function2(obj)
    function3()

def test_myfunction:
    function1_mock.assert_called_once_with()
    function2_mock.assert_called_once_with(function1_mock.return_value)
    function3_mock.assert_called_once_with()
```

As for me, it looks like we wrote the same code twice :-) 

Mocks help sometimes, of course. I just don't think it's the case here

Reply:
ah I think I understand what you're saying, we might be agreeing! In this test I am not actually mocking the implementation of `quote`, since I'm using `wraps`:
```
mock.Mock(wraps=quote_original)
```

So calling the "mock" in this case still calls the original, the only purpose of the mock is so that we can assert that it was called (and how it was called, and how many times).

Does that make sense? I can add comments in the code to help make this clearer for readers.

Let me know if you still want to remove; if so could you show me what you'd like the asserts to look like?

Reply:
```python
self.assertEqual(s, "baz=bar;baz=quux;foo=asdf")
```

Reply:
I have completely removed the wrapped calls in https://github.com/devopshq/artifactory/pull/409/commits/2f1551c439924ecd0537ea01d806f9686607146c as requested.